contracts/Comptroller/Diamond/Diamond.sol

```diff
@@ -0,0 +1,200 @@
+pragma solidity 0.5.16;
+pragma experimental ABIEncoderV2;
+
+import { IDiamondCut } from "./interfaces/IDiamondCut.sol";
+import "../ComptrollerStorage.sol";
+import "../Unitroller.sol";
+
+contract Diamond is ComptrollerV12Storage {
+    event DiamondCut(IDiamondCut.FacetCut[] _diamondCut);
+
+    /**
+     * @notice Call _acceptImplementation to accept the diamond proxy as new implementaion.
+     * @param unitroller Address of the unitroller.
+     */
+    function _become(Unitroller unitroller) public {
+        require(msg.sender == unitroller.admin(), "only unitroller admin can");
+        require(unitroller._acceptImplementation() == 0, "not authorized");
+    }
+
+    /**
+     * @notice To add function selectors to the facets' mapping.
+     * @param _diamondCut IDiamondCut contains facets address, action and function selectors.
+     */
+    function diamondCut(IDiamondCut.FacetCut[] memory _diamondCut) public {
+        require(msg.sender == admin, "only unitroller admin can");
+        libDiamondCut(_diamondCut);
+    }
+
+    /**
+     * @notice To add function selectors to the facets' mapping.
+     * @param _diamondCut IDiamondCut contains facets address, action and function selectors.
+     */
+    function libDiamondCut(IDiamondCut.FacetCut[] memory _diamondCut) internal {
+        for (uint256 facetIndex; facetIndex < _diamondCut.length; facetIndex++) {
+            IDiamondCut.FacetCutAction action = _diamondCut[facetIndex].action;
+            if (action == IDiamondCut.FacetCutAction.Add) {
+                addFunctions(_diamondCut[facetIndex].facetAddress, _diamondCut[facetIndex].functionSelectors);
+            } else if (action == IDiamondCut.FacetCutAction.Replace) {
+                replaceFunctions(_diamondCut[facetIndex].facetAddress, _diamondCut[facetIndex].functionSelectors);
+            } else if (action == IDiamondCut.FacetCutAction.Remove) {
+                removeFunctions(_diamondCut[facetIndex].facetAddress, _diamondCut[facetIndex].functionSelectors);
+            } else {
+                revert("LibDiamondCut: Incorrect FacetCutAction");
+            }
+        }
+        emit DiamondCut(_diamondCut);
+    }
+
+    /**
+     * @notice Add function selectors to the facet's address mapping.
+     * @param _facetAddress Address of the facet.
+     * @param _functionSelectors Array of function selectors need to add in the mapping.
+     */
+    function addFunctions(address _facetAddress, bytes4[] memory _functionSelectors) internal {
```

Comment:
Discussion point (no need to change anything in this PR): can we settle on a single naming convention for parameters, internal functions and private state variables? e.g., `_param` vs `param_` vs `param`, `internalFunction` vs `_internalFunction`, `privateStateVar` vs `_privateStateVar`.

cc @chechu @narayanprusty @0xlucian @GitGuru7

Reply:
Mu opinions are similar as @0xlucian